sound/soc/sof/intel/hda-codec.c

```diff
@@ -151,6 +151,8 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address,
 		if (!hdev->bus->audio_component) {
 			dev_dbg(sdev->dev,
 				"iDisp hw present but no driver\n");
+
+			snd_hdac_ext_bus_device_exit(hdev);
```

Comment:
it would help to re-explain in your comment message that when hda_codec_probe() returns an error, we disable the codec and keep going. Returning -ENOENT does not stop the probe, so your problem statement is that we keep going but have not released the resources that will be ignored.

Reply:
thanks!